packages/Main/test/unit/vpc.js

```
@@ -48,30 +74,30 @@ describe('VPC', function () {
                         assert.ok(vpcCopcSource.sources.length > 0);
                         done();
                     }).catch(done);
-            }).timeout(5000);
+            });
         });
+    });
 
-        describe('stack sources', function (done) {
-            it('ept stack', function () {
-                const eptMockSource = vpcEptSource.sources[0];
-                vpcEptSource.instantiate(eptMockSource);
+    describe('Stacked sources', function (done) {
+        it('instantiate ept stacked source', function () {
+            const eptMockSource = vpcEptSource.sources[0];
+            vpcEptSource.instantiate(eptMockSource);
 
-                eptMockSource.whenReady
-                    .then(() => {
-                        assert.ok(vpcEptSource.sources[0].isEntwinePointTileSource);
-                        done();
-                    }).catch(done);
-            });
-            it('copc stack', function () {
-                const copcMockSource = vpcCopcSource.sources[0];
-                vpcCopcSource.instantiate(copcMockSource);
+            eptMockSource.whenReady
+                .then(() => {
+                    assert.ok(vpcEptSource.sources[0].isEntwinePointTileSource);
+                    done();
+                }).catch(done);
+        });
+        it('instanciated copc stacked source', function () {
+            const copcMockSource = vpcCopcSource.sources[0];
+            vpcCopcSource.instantiate(copcMockSource);
 
-                copcMockSource.whenReady
-                    .then(() => {
-                        assert.ok(vpcCopcSource.sources[0].isCopcSource);
-                        done();
-                    }).catch(done);
-            });
+            copcMockSource.whenReady
+                .then(() => {
+                    assert.ok(vpcCopcSource.sources[0].isCopcSource);
+                    done();
+                }).catch(done);
```

I think the catched error may be lost with `catch(done)`. Shouldn't we do something like this instead ? :
```
.catch((err) => {
    console.error(err);
    done(err);
}); 
```

---

If I'm not worng:
``
.catch(done);
``
is the same as 
``
.catch((err) => done(err));
``
what is the same as what you advice, without the console.error().

I tried to generate errors on purpose in:
- the vpc.js file
- vpSource.instantiate()
- in the assert.ok()
- in the copcSource.whenReady (for mockSource.whenReady)

And all errors are well catch by Mocha, what is in my opinion what we aims in the units tests.


But I did forgot a 'done' as argument of the it function for the 'instanciated copc stacked source'...
So all that tests were not for nothing ;-)

---

:grin:  ok thanks for the tests and the feedback :+1: